src/test/java/core/basesyntax/PairTest.java

```diff
@@ -0,0 +1,56 @@
+package core.basesyntax;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.lang.reflect.Method;
+
+public class PairTest {
+
+    private Pair firstPair;
+    private Pair secondPair;
+    private Pair thirdPair;
+
+    @Before
+    public void init() {
+        firstPair = Pair.of(5, "Five");
+        secondPair = Pair.of(2, "Two");
+        thirdPair = Pair.of(5, "Five");
+    }
+
+    @Test
+    public void methodsCheck() {
+        Class<Pair> pairClass = Pair.class;
+        Method[] methods = pairClass.getMethods();
+        Assert.assertTrue("Not all methods were implemented", methods.length >= 5);
+    }
+
+    @Test
+    public void pairsAreEquals() {
+        Assert.assertEquals(firstPair, thirdPair);
+    }
+
+    @Test
+    public void pairsAreNotEquals() {
+        Assert.assertNotEquals(firstPair, secondPair);
+    }
+
+    @Test
+    public void hashCodeTest() {
+        Assert.assertEquals("Hash codes are not equal!", firstPair.hashCode(), thirdPair.hashCode());
+        Assert.assertNotEquals("Hash codes are not equal!", firstPair.hashCode(), secondPair.hashCode());
```

Comment:
```suggestion
        Assert.assertNotEquals("Hash codes should not equal!", firstPair.hashCode(), secondPair.hashCode());
```
please pay more attention on what you are copy pasting